internal/pkg/deploy/cloudformation/app.go

```diff
@@ -49,23 +49,28 @@ func (cf CloudFormation) DeployApp(template, stackName, changeSetName string) er
 		StackName:     aws.String(stackName),
 		TemplateBody:  aws.String(template),
 		Capabilities:  aws.StringSlice([]string{cloudformation.CapabilityCapabilityIam}),
-		ChangeSetType: aws.String("UPDATE"),
+		ChangeSetType: aws.String(cloudformation.ChangeSetTypeUpdate),
 	})
 
 	if err != nil {
 		// TODO: gracefully handle stack already in UPDATE_IN_PROGRESS state
 		return fmt.Errorf("create change set: %w", err)
 	}
 
-	err = cf.client.WaitUntilChangeSetCreateComplete(&cloudformation.DescribeChangeSetInput{
+	describeChangeSetInput := &cloudformation.DescribeChangeSetInput{
 		ChangeSetName: aws.String(changeSetName),
 		StackName:     aws.String(stackName),
-	})
+	}
+
+	err = cf.client.WaitUntilChangeSetCreateComplete(describeChangeSetInput)
 
-	// NOTE: WaitUntilChangeSetCreateComplete just straight up fails if there are no changes to apply.
 	if err != nil {
-		// TODO: find a better way to handle no ChangeSet changes.
-		// Chances are here that there are no changes to apply, but it's janky to handle.
+		out, err := cf.client.DescribeChangeSet(describeChangeSetInput)
```

Comment:
This will overwrite the error from `WaitUntilChangeSetCreateComplete`, do we want that?

Reply:
Mmm good catch. That was not intentional, but it's probably fine since `WaitUntilChangeSetCreateComplete` makes `DescribeChangeSet` calls, so the error should theoretically be the same?

Reply:
Gotcha, if this call errors out would `out.Changes` also result in a segfault?

Maybe we should declare both errors separately and check if its not nil?

Reply:
That's true, I'm not handling the `DescribeChangeSet` error properly, will fix.